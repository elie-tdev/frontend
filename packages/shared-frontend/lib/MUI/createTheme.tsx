import { createTheme as createMuiTheme } from '@mui/material/styles'

import { makeDefaultTheme } from './defaultTheme'
import { BAIThemeOptions } from './types'
import { createDefaultPaletteOptions } from './utils/palette'
import { createDefaultSpacingOptions } from './utils/spacing'

const createThemeBase = (themeOptions: BAIThemeOptions, isTest = false) => {
  createDefaultSpacingOptions(themeOptions)
  createDefaultPaletteOptions(themeOptions)
  return makeDefaultTheme(createMuiTheme(themeOptions), !isTest)
}

export const createTheme = (themeOptions: BAIThemeOptions = {}) =>
  createThemeBase(themeOptions)

export const createThemeTest = (themeOptions: BAIThemeOptions = {}) =>
  createThemeBase(themeOptions, true)
