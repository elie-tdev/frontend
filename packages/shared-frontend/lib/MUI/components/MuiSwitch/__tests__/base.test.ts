import { MuiSwitch } from '../'
import { createThemeTest } from '../../../createTheme'

describe('Switch', () => {
  const theme = createThemeTest()
  const component = MuiSwitch(theme)

  it('should be consistent with previous versions: ', () => {
    expect(component).toMatchSnapshot()
  })
})
