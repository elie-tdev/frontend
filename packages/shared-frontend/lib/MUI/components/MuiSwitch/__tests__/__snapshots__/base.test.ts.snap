// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should be consistent with previous versions:  1`] = `
Object {
  "defaultProps": Object {
    "focusRipple": false,
  },
  "styleOverrides": Object {
    "colorPrimary": Object {
      "&.Mui-checked + .MuiSwitch-track": Object {
        "color": "#1B76D4",
        "opacity": 1,
      },
    },
    "colorSecondary": Object {
      "&.Mui-checked + .MuiSwitch-track": Object {
        "color": "#757575",
        "opacity": 1,
      },
    },
    "input": Object {
      "backgroundColor": "transparent",
    },
    "root": Object {
      "color": "#bdbdbd",
      "contain": "strict",
      "height": 32,
      "margin": 8,
      "padding": 0,
      "width": 52,
    },
    "sizeSmall": Object {
      "& .MuiSwitch-switchBase": Object {
        "&.Mui-checked": Object {
          "transform": "translateX(12px)",
        },
        "&:hover": Object {
          "backgroundColor": "transparent",
        },
        "margin": -2,
        "padding": 4,
      },
      "& .MuiSwitch-thumb": Object {
        "height": 12,
        "width": 12,
      },
      "& .MuiSwitch-track": Object {
        "borderRadius": 8,
      },
      "height": 16,
      "width": 28,
    },
    "switchBase": Object {
      "&.Mui-checked": Object {
        "& + &.MuiSwitch-track": Object {
          "backgroundColor": "currentColor",
          "border": "none",
          "opacity": 1,
        },
        "transform": "translateX(20px)",
      },
      "&.Mui-disabled": Object {
        "& .MuiSwitch-thumb": Object {
          "boxShadow": "none",
        },
        "&.Mui-checked": Object {
          "opacity": 0.4,
        },
        "&.Mui-checked + .MuiSwitch-track, & + .MuiSwitch-track": Object {
          "border": "none",
          "opacity": 0.2,
        },
        "backgroundColor": "none",
        "border": "none",
        "opacity": 0.4,
      },
      "&.Mui-focusVisible": Object {
        "& + .MuiSwitch-track": Object {
          "boxShadow": "inset 0px 0px 0px 1px #bdbdbd, inset 0px 0px 0px 3px #FFFFFF",
        },
        "& .MuiSwitch-thumb": Object {
          "boxShadow": "none",
        },
      },
      "&.Mui-focusVisible.Mui-checked": Object {
        "&.MuiSwitch-colorPrimary": Object {
          "& + .MuiSwitch-track": Object {
            "boxShadow": "inset 0px 0px 0px 1px #1B76D4, inset 0px 0px 0px 3px #FFFFFF",
          },
        },
        "&.MuiSwitch-colorSecondary": Object {
          "& + .MuiSwitch-track": Object {
            "boxShadow": "inset 0px 0px 0px 1px #757575, inset 0px 0px 0px 3px #FFFFFF",
          },
        },
      },
      "padding": 3,
    },
    "thumb": Object {
      "backgroundColor": "white",
      "border": "inset 2px solid #bdbdbd",
      "height": 26,
      "width": 26,
    },
    "track": Object {
      "backgroundColor": "currentColor",
      "borderRadius": 16,
      "opacity": 1,
      "transition": "background-color 200ms ease, border 200ms ease",
      "willChange": "background-color, border, box-shadow",
    },
  },
}
`;
